scripts/performMapping.sh

```diff
@@ -19,34 +19,58 @@ while getopts ":i:o:m:g:b:" o; do
         b)
             BATCHSIZE=${OPTARG}
             ;;
+        j)
+            SERVER_URL=${OPTARG}
+            ;;
         *)
             usage
             ;;
     esac
 done
 shift $((OPTIND-1))
 
-echo "Mapping Records"
-numfiles=$(find $RECORDSINPUTFOLDER -type f -name '*.xml' | wc -l)
-count=1
-echo "Found $numfiles record XML files in $RECORDSINPUTFOLDER"
-
-(
-for f in $(find $RECORDSINPUTFOLDER -type f -name '*.xml' ); do
-  ((i=i%BATCHSIZE)); ((i++==0)) && wait
-  echo "Mapping record $count of $numfiles ($f)"
-  o=${f/.xml/.ttl}
-  o=${o/$RECORDSINPUTFOLDER/}
-  java --add-opens java.base/java.lang.reflect=ALL-UNNAMED \
-    --add-opens java.base/java.util=ALL-UNNAMED \
-    --add-opens java.base/java.text=ALL-UNNAMED \
-    --add-opens java.desktop/java.awt.font=ALL-UNNAMED \
-    -jar /x3ml/x3ml-engine.exejar \
-    --input $f \
-    --x3ml $RECORDMAPPING \
-    --policy $GENERATOR \
-    --output $RECORDSOUTPUTFOLDER/$o \
-    --format text/turtle &
-  count=$((count+1)) 
+process_file() {
+    local file="$1"
+    local output_file="$RECORDSOUTPUTFOLDER/$(basename "$file" .xml).ttl"
+    echo
+    echo curl -X POST "$SERVER_URL?mappingFile=$RECORDMAPPING&generatorPolicy=$GENERATOR&inputFile=$file&outputFile=$output_file"
+    curl -X POST "$SERVER_URL?mappingFile=$RECORDMAPPING&generatorPolicy=$GENERATOR&inputFile=$file&outputFile=$output_file"
+    echo
+}
+
+export -f process_file
+export SERVER_URL
+export RECORDSOUTPUTFOLDER
+export RECORDMAPPING
+export GENERATOR
+
+# process a batch of files
+process_batch() {
+    batch=("$@")
+    echo "Processing batch of ${#batch[@]} files:"
+    for file in "${batch[@]}"; do
+        process_file "$file"
+    done
+}
+
+batch=()
+
+for file in "$RECORDSINPUTFOLDER"/*.xml; do
+    if [ -f "$file" ]; then
+        batch+=("$file")
+
+        # Process the batch if we reach the batch size
+        if [ "${#batch[@]}" -eq "$BATCHSIZE" ]; then
+            process_batch "${batch[@]}"
+            batch=() # Reset the batch
+        fi
+    else
+        echo "No XML files found in the directory."
+        break
+    fi
 done
```

Comment:
I originally added the batch processing to have several java engines spin up simultaneously, i.e. as a basic form of optimisation. I think with the java service, batch processing is no longer necessary. Can you change it so that files are processed individually?

Reply:
changed so that files are processed individually